docs/plots/backends.rst

```diff
@@ -0,0 +1,362 @@
+************************
+Plotting backend details
+************************
+
+.. warning::
+   The plotting backend API is currently re-designed and details might change in the near future.
+
+Plotting in Sherpa is done through *plotting backends*. A *backend* is an
+external package that performs the actual plotting function such as :term:`matplotlib`, i.e. it creates a
+canvas and puts color on paper or on the screen. The Sherpa plotting classes are
+written in a backend-independent way; they work with any plotting backend that
+is supported by Sherpa. We also use the term *backend* to refer to Sherpa
+classes that connect the Sherpa plot objects with the plotting backend by
+translating Sherpa options into backend-specific commands. The details of that
+are explained below, but, in short, a backend class would translate the sherpa
+backend-independent "plot a line in red from here to there" into backend
+specific ``plt.plot(x,y, linecolor='r')`` (if the backend is :term:`matplotlib`).
+
+Instead of utilizing Sherpa classes and commands to plot data, users can
+alternatively just access the data in the Sherpa data objects (e.g. the x and y
+values of a dataset) and perform plotting operations directly with any plotting
+backend that is available to them. This method may be less convenient, but it
+works for ploting backends not (yet) supported by Sherpa.
+
+Which backend is used?
+======================
+
+When the `sherpa.plot` module is first imported, Sherpa tries to import the
+backends installed with Sherpa in the order listed in the
+``options.plot_pkg`` setting from the ``sherpa.rc`` startup file.
+The first module that imports successfully is set as the active
+backend. The following command prints the name and the location
+on disk of that module::
+
+   >>> from sherpa import plot
+   >>> print(plot.backend)
+
+Change the backend
+------------------
+
+.. warning::
+   When the plotting backend is changed, new plotting objects must be
+   initialized because objects like e.g. `sherpa.plot.DataPlot` keep a
+   copy of all default parameters of the backend that was active when they
+   are initialized. A different backend might not understand all those
+   parameters.
+
+   In particular, that chnging the backend might not work for plot functions
+   in Sherpa's UI layer such as `sherpa.astro.plot.plot_data` because those
+   work with objects underneath that are not updated when the plotting backend
+   is changed.
+
```

Comment:
There's a typo here - `changing` - but I think the text is not clear. It also needs to reference something like `sherpa.astro.ui.plot_data` instead of `sherpa.astro.plot.plot_data` but I can't guarantee that'll work.